Incorrect index signature for NbAclRole

### Issue type

**I'm submitting a ...**

* [x] bug report
* [ ] feature request

### Issue description
NbAclRole is defined incorrectly, which in turn causes problems with strict projects. As of now it has an index signature as `[permission: string]: string|string[]` but it also defines `parent?: string` which doesn't conform to that signature.

**Current behavior:**
NbAclRole has an invalid index signature and causes compilation problems when strict mode is enforced.

**Expected behavior:**
NbAclRole should have a valid signature.

**Steps to reproduce:**
Doesn't apply.

**Related code:**
```
export interface NbAclRole {
  parent?: string,
  [permission: string]: string|string[],
}
```

### Other information:

**npm, node, OS, Browser**
```
Node: v10.14.1
npm: 6.4.1
OS: Windows 10
```

**Angular, Nebular**
```
Angular: 7.1.3
Nebular: 3.0.1 but also applies to 3.1.0 and master
```

Hi @nnixaa, here you go:

I've created a minimal test repo since i can't reproduce with stackblitz (i'm guessing it doesn't report errors the way i expected): https://github.com/idsplus-carlos/issue_nebular_1166

The error is as follows:
```
Failed to compile.

node_modules/@nebular/security/security.options.d.ts(3,5): error TS2411: Property 'parent' of type 'string | undefined' is not assignable to string index type 'string | string[]'.
```

Repro steps:

1. ng new testproject && cd testproject
1. npm i @nebular/security
1. Add strict: true to compilerOptions in /tsconfig.json
1. Import NbSecurityModule

--- a/src/framework/security/security.options.ts
+++ b/src/framework/security/security.options.ts
@@ -2,7 +2,7 @@ import { InjectionToken } from '@angular/core';
 
 export interface NbAclRole {
   parent?: string,
-  [permission: string]: string|string[],
+  [permission: string]: string|string[]|undefined,
 }
 
 export interface NbAccessControl {

